docs/SpellExample.java

```diff
@@ -0,0 +1,315 @@
+        ).get();
+
+        String key = keyResults.at("secret").to(String.class).get();
+        FaunaClient client = adminClient.newSessionClient(key);
+        System.out.println("Connected to Fauna database " + DB_NAME + " with server role\n");
+        adminClient.close();
+
+
+        /*
+         * Create the spell class and index
+         */
+        String SPELLS_CLASS = "spells";
+        String INDEX_NAME = "spells_index";
+
+        Value classResults = client.query(
+            CreateClass(
+                Obj("name", Value(SPELLS_CLASS))
+            )
+        ).get();
+        System.out.println("Create Class for " + DB_NAME + ":\n " + toPrettyJson(classResults) + "\n");
+
+        Value indexResults = client.query(
+            CreateIndex(
+                Obj("name", Value(INDEX_NAME), "source", Class(Value(SPELLS_CLASS))
+                )
+            )
+        ).get();
+        System.out.println("Create Index for " + DB_NAME + ":\n " + toPrettyJson(indexResults) + "\n");
+
+
+        /*
+         * Add some entries to the spells class
+         */
+
+        Value addFireResults = client.query(
+            Create(
+                Class(Value(SPELLS_CLASS)),
+                Obj("data",
+                    Obj("name", Value("Fire Beak"), "element", Value("water"), "cost", Value(15))
+                )
+            )
+        ).get();
+        System.out.println("Added spell to class " + SPELLS_CLASS + ":\n " + toPrettyJson(addFireResults) + "\n");
+
+
+        Value addDragonResults = client.query(
+            Create(
+                Class(Value(SPELLS_CLASS)),
+                Obj("data",
+                    Obj("name", Value("Water Dragon's Claw"), "element", Value("water"), "cost", Value(25))
+                )
+            )
+        ).get();
+        System.out.println("Added spell to class " + SPELLS_CLASS + ":\n " + toPrettyJson(addDragonResults) + "\n");
+
+        Value addHippoResults = client.query(
+            Create(
+                Class(Value(SPELLS_CLASS)),
+                Obj("data",
+                    Obj("name", Value("Hippo's Wallow"), "element", Value("water"), "cost", Value(35))
+                )
+            )
+        ).get();
+        System.out.println("Added spell to class " + SPELLS_CLASS + ":\n " + toPrettyJson(addHippoResults) + "\n");
+
+        //The results at 'ref' are are a resource pointer to the class that was just created.
+        Value hippoRef = addHippoResults.at("ref");
+        System.out.println("hippoRef = " + hippoRef);
+
+        /*
+         * Read the hippo back that we just created
+         */
+
+        Value getHippoResults = client.query(
+            Select(Value("data"),Get(hippoRef))
+        ).get();
+        System.out.println("Hippo Spell:\n " + toPrettyJson(getHippoResults) + "\n");
+
+        //convert the hippo results into primitive elements
+        String element = getHippoResults.get(Field.at("element").to(String.class));
```

Comment:
The `Field` extractors are little trick to understand at first. How about:
```java
getHippoResults.at("element").to(String.class).get();
```

Reply:
Yes they are tricky!  Is there any comments we would want to add to the docs about why one way vs. the other?